### ISSUE TITLE
Add hyperparameter tuning to episode 3

### ISSUE THREAD

**Issue by svenvanderburg** [carpentries-incubator/deep-learning-intro#351]
Add explanation of hyperparamter tuning to episode 3

In this section: https://carpentries-incubator.github.io/deep-learning-intro/03-monitor-the-model/index.html#counteract-model-overfitting we are actually doing hyperparamter tuning (we are changing a hyperparameter). It is only a small extra step to do this within a for-loop. 

- [ ] ~~Add an infobox that discusses hyperparameter tuning in a for-loop~~ see https://github.com/carpentries-incubator/deep-learning-intro/issues/351#issuecomment-1621589631
- [ ] Have a callout to explain grid search.

You can get inspiration from @qualiaMachine 's script here: https://colab.research.google.com/drive/1B_F5ik6P9hfbKT2pzfMgsvFeF99_Uy9V?usp=sharing

This issue resulted from discussions in #348

**Comment by svenvanderburg** [comment]
I just realized we already have something like this in the final exercise related to dropout: https://carpentries-incubator.github.io/deep-learning-intro/04-advanced-layer-types/index.html#vary-dropout-rate We can slightly build that out further by adding an infobox on grid search.
